hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/StreamWriteOperatorCoordinator.java

```diff
@@ -349,8 +351,8 @@ private static void initMetadataTable(HoodieFlinkWriteClient<?> writeClient) {
     writeClient.initMetadataTable();
   }
 
-  private static CkpMetadata initCkpMetadata(HoodieTableMetaClient metaClient, Configuration conf) throws IOException {
-    CkpMetadata ckpMetadata = CkpMetadata.getInstance(metaClient, conf.getString(FlinkOptions.WRITE_CLIENT_ID));
+  private static CkpMetadata initCkpMetadata(HoodieTableMetaClient metaClient, HoodieWriteConfig writeConfig, Configuration conf) throws IOException {
+    CkpMetadata ckpMetadata = CkpMetadataFactory.getCkpMetadata(metaClient, writeConfig, conf);
     ckpMetadata.bootstrap();
     return ckpMetadata;
```

Comment:
When the instant state was updated ?

Reply:
So this action must be succeed, or the writers would be blocked infinitely, how could we ensure that?

Reply:
You are right, we can't ensure that the refreshing by the coordinator is successful.  I added a new mechanism that the instant states will be refresh if client request count reaches a threshold, with this mechanism we can ensure that the writer won't be blocked infinitely.